# **Walmart's Inventory Management and Demand Forecasting**

*Our objective is to create a precise sales prediction model for diverse Walmart stores, incorporating variables like date, store category, promotions, and environmental data. The model's predictions will be applied to fine-tune inventory levels, mitigating the risk of excess stock or shortages that could impact product availability. This optimization aims to improve store operational efficiency and enhance overall customer satisfaction.*

*This notebook will particularly focus on the model's inference process.*

## **I. Introduction**

### **`Members:`**
- Faris Arief Mawardi
- Michael Nathaniel
- Nadia Nabilla Shafira
- Noufal Rifata Reyhan

### **`Objective:`**

Evaluate the sales forecasting model's performance for various Walmart stores using dummy data. Utilize factors such as date, store type, promotions, and environmental information to analyze how well the model predicts sales outcomes. The examination aims to assess the model's accuracy and effectiveness in optimizing inventory levels, thereby preventing stock excesses or shortages. This process is essential for ensuring seamless product availability, improving store operational efficiency, and enhancing overall customer satisfaction.

# **II. Import Libraries**

In [2]:
import pandas as pd
import numpy as np

import pickle

---
# **III. Load Model**
> *In this step, the pre-trained or previously saved sales forecasting model is retrieved and loaded into the current environment. Loading the model is a crucial phase in the workflow, as it allows the subsequent steps to utilize the learned patterns and information embedded in the model during its training phase. Once the model is successfully loaded, it becomes ready for deployment and can be employed to make predictions or inferences on new data, enabling the assessment of its accuracy and performance in forecasting sales based on the provided features such as store type, promotions, and environmental factors.*

In [3]:
with open('best_rf.pkl', 'rb') as model:
    rf_model = pickle.load(model)

---
# **IV. Data Dummy**
> *In this step, dummy data is introduced into the system for the purpose of testing and validating the loaded sales forecasting model. Dummy data refers to simulated or placeholder information that mimics the structure and format of real-world data but may not necessarily represent authentic observations. The utilization of dummy data allows for a controlled environment to assess how well the model performs in making predictions and generating forecasts.*

In [14]:
dataInf = {
    'Store' : [12, 7, 23],
    'Temperature' : [61.2, 58.91, 47.74],
    'Fuel_Price' : [3.411, 2.453, 3.286],
    'MarkDown1' : [7621, 10224, 9820],
    'MarkDown2' : [3752, 2683, 1291],
    'MarkDown3' : [4614, 11942, 1329],
    'MarkDown4' : [421, 769, 135],
    'MarkDown5' : [9921, 3571, 1231],
    'CPI' : [217.1, 632.5, 382.7],
    'Unemployment' : [6.8, 3.1, 8.7],
    'IsHoliday' : ['True', 'True', 'False'],
    'Type' : ['A', 'C', 'B'],
    'Size' : [231192, 98421, 119521],
    'Dept' : [28, 91, 36],
}

dataInf = pd.DataFrame(dataInf)
dataInf

Store  Temperature  Fuel_Price  MarkDown1  MarkDown2  MarkDown3  MarkDown4  \
0     12        61.20       3.411       7621       3752       4614        421   
1      7        58.91       2.453      10224       2683      11942        769   
2     23        47.74       3.286       9820       1291       1329        135   

   MarkDown5    CPI  Unemployment IsHoliday Type    Size  Dept  
0       9921  217.1           6.8      True    A  231192    28  
1       3571  632.5           3.1      True    C   98421    91  
2       1231  382.7           8.7     False    B  119521    36

In [15]:
dataInf = dataInf.drop(['Unemployment', 'IsHoliday'], axis = 1)

In [16]:
dataInf

Store  Temperature  Fuel_Price  MarkDown1  MarkDown2  MarkDown3  MarkDown4  \
0     12        61.20       3.411       7621       3752       4614        421   
1      7        58.91       2.453      10224       2683      11942        769   
2     23        47.74       3.286       9820       1291       1329        135   

   MarkDown5    CPI Type    Size  Dept  
0       9921  217.1    A  231192    28  
1       3571  632.5    C   98421    91  
2       1231  382.7    B  119521    36

---
# **IV. Prediction**
> *In this step, the loaded sales forecasting model is applied to the dummy data to generate predictions. The model utilizes the information embedded during its training phase, including patterns learned from historical sales data, store types, promotional activities, and environmental factors. By inputting the dummy data into the model, it calculates and produces forecasts for sales outcomes based on the provided features.*

In [17]:
yPred_inf = rf_model.predict(dataInf)
yPred_inf

array([1247.74285, 6854.859  , 1416.30115])

From the outcomes provided, it is evident that the model anticipates sales to be 1247.74 for index 0, 6854.85 for index 1, and 1416.30 for index 2.